src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/serviceprovider/ReferralReportJobListener.kt

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.reporting.serviceprovider
+
+import mu.KLogging
+import net.logstash.logback.argument.StructuredArguments
+import org.springframework.batch.core.BatchStatus
+import org.springframework.batch.core.JobExecution
+import org.springframework.batch.core.JobExecutionListener
+import org.springframework.stereotype.Component
+import java.io.File
+import kotlin.io.path.createTempDirectory
+
+@Component
+class ReferralReportJobListener() : JobExecutionListener {
+  companion object : KLogging()
+
+  override fun beforeJob(jobExecution: JobExecution) {
+    // create a temp file for the job to write to and store the path in the execution context
+    val params = jobExecution.jobParameters.parameters
```

Comment:
I think validation of the parameters would be useful

Reply:
nice idea - how would you validate them?

Reply:
Perhaps in the beforeJob and check that params that are defined in Config and in the Listener are not null?